lib/engine-native/src/engine.rs

```diff
@@ -27,6 +29,10 @@ impl NativeEngine {
     /// Create a new `NativeEngine` with the given config
     #[cfg(feature = "compiler")]
     pub fn new(compiler: Box<dyn Compiler + Send>, target: Target, features: Features) -> Self {
+        if let Err(_) = which("gcc").or_else(|_| which("clang-10")) {
+            panic!("gcc/clang is not found, it is required for using the `NativeEngine`");
+        }
```

Comment:
Once we know which compiler to use, we should save it in the struct, along with if supports cross_compilation or not.

To then, use that command in the artifact generation.